src/status_im/contexts/wallet/wallet_connect/responding_events.cljs

```diff
@@ -0,0 +1,96 @@
+             constants/wallet-connect-eth-sign-typed-v4-method
+             [:dispatch [:wallet-connect/respond-sign-typed-data password :v4]])]})))
+
+(rf/reg-event-fx
+ :wallet-connect/respond-eth-sign
+ (fn [{:keys [db]} [password]]
+   (let [{:keys [address raw-data]} (get db :wallet-connect/current-request)]
+     {:fx [[:effects.wallet-connect/sign-message
+            {:password   password
+             :address    address
+             :data       raw-data
+             :on-fail    #(rf/dispatch [:wallet-connect/on-sign-failed %])
+             :on-success #(rf/dispatch [:wallet-connect/send-response %])}]]})))
+
+
+(rf/reg-event-fx
+ :wallet-connect/respond-personal-sign
+ (fn [{:keys [db]} [password]]
+   (let [{:keys [address raw-data]} (get db :wallet-connect/current-request)]
+     {:fx [[:effects.wallet-connect/sign-message
+            {:password   password
+             :address    address
+             :data       raw-data
+             :on-fail    #(rf/dispatch [:wallet-connect/on-sign-failed %])
+             :on-success #(rf/dispatch [:wallet-connect/send-response %])}]]})))
+
+(rf/reg-event-fx
+ :wallet-connect/respond-sign-typed-data
+ (fn [{:keys [db]} [password typed-data-version]]
+   (let [{:keys [address raw-data]} (get db :wallet-connect/current-request)]
+     {:fx [[:effects.wallet-connect/sign-typed-data
+            {:password   password
+             :address    address
+             :data       raw-data
+             :version    typed-data-version
+             :on-fail    #(rf/dispatch [:wallet-connect/on-sign-failed %])
+             :on-success #(rf/dispatch [:wallet-connect/send-response %])}]]})))
+
+(rf/reg-event-fx
+ :wallet-connect/on-sign-failed
+ (fn [{:keys [db]} [error]]
+   (let [event                      (get-in db [:wallet-connect/current-request :event])
+         {:keys [raw-data address]} (get db :wallet-connect/current-request)
+         method                     (wallet-connect-core/event->method event)]
+     (log/error "Failed to sign Wallet Connect request"
+                {:error                error
+                 :address              address
+                 :sign-data            raw-data
+                 :method               method
+                 :wallet-connect-event event
+                 :event                :wallet-connect/on-sign-failed})
+     {:fx [[:dispatch [:wallet-connect/close-session-request]]]})))
+
+
+(rf/reg-event-fx
+ :wallet-connect/send-response
+ (fn [{:keys [db]} [result]]
+   (let [{:keys [id topic] :as event} (get-in db [:wallet-connect/current-request :event])
+         method                       (wallet-connect-core/event->method event)
+         web3-wallet                  (get db :wallet-connect/web3-wallet)]
+     {:fx [[:effects.wallet-connect/respond-session-request
+            {:web3-wallet web3-wallet
+             :topic       topic
+             :id          id
+             :result      result
+             :on-fail     (fn [error]
```

Comment:
A minor detail: it would be slightly better if the name was `on-error` just because that's what we use in `json-rpc/call`. I've seen a bug a few months ago being fixed in develop where the developer used `on-fail` incorrectly in `json-rpc/call` instead of `on-error`. Although we can catch such problems with malli, it would be good if we used a consistent name for these success/error callbacks.